server/src/main/java/io/crate/metadata/sys/SysClusterHealth.java

```diff
@@ -0,0 +1,103 @@
+import org.elasticsearch.cluster.block.ClusterBlock;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.rest.RestStatus;
+
+import io.crate.metadata.RelationName;
+import io.crate.metadata.SystemTable;
+
+public class SysClusterHealth {
+
+    public static final RelationName IDENT = new RelationName(SysSchemaInfo.NAME, "cluster_health");
+
+    static SystemTable<ClusterHealth> INSTANCE = SystemTable.<ClusterHealth>builder(IDENT)
+        .add("health", STRING, ClusterHealth::getHealth)
+        .add("severity", SHORT, ClusterHealth::getSeverity)
+        .add("description", STRING, ClusterHealth::description)
+        .add("missing_shards", LONG, ClusterHealth::missingShards)
+        .add("underreplicated_shards", LONG, ClusterHealth::underreplicatedShards)
+        .build();
+
+    public static CompletableFuture<Iterable<ClusterHealth>> compute(ClusterState clusterState) {
+        // Following implementation of {@link org.elasticsearch.cluster.health.ClusterStateHealth}
+        Set<ClusterBlock> blocksRed = clusterState.blocks().global(RestStatus.SERVICE_UNAVAILABLE);
+        if (!blocksRed.isEmpty()) {
+            var block = blocksRed.iterator().next();
+            ClusterHealth clusterHealth = new ClusterHealth(
+                TableHealth.Health.RED,
+                block.description(),
+                -1,
+                -1
+            );
+            return CompletableFuture.completedFuture(List.of(clusterHealth));
+        }
+        Set<ClusterBlock> blocksYellow = clusterState.blocks().global(ClusterBlockLevel.METADATA_WRITE);
+        final TableHealth.Health clusterHealth = !blocksYellow.isEmpty() ? TableHealth.Health.YELLOW : TableHealth.Health.GREEN;
+        final String description = !blocksYellow.isEmpty() ? blocksYellow.iterator().next().description() : "";
+        return TableHealth.compute(clusterState)
+            .thenApply((it) -> {
+                long missingShards = 0;
+                long underreplicatedShards = 0;
+                TableHealth.Health health = clusterHealth;
+                String finalDescription = description;
+                for (var tableHealth : it) {
+                    if (tableHealth.health().severity() > health.severity()) {
+                        health = tableHealth.health();
+                        // shard level health is only set to RED if there are missing primary shards that were
+                        // allocated before see {@link ClusterShardHealth#getInactivePrimaryHealth()}.
+                        // Otherwise, the table health is set to YELLOW.
+                        if (health == TableHealth.Health.RED || tableHealth.getMissingShards() > 0) {
+                            finalDescription = "One or more tables are missing shards";
+                        } else if (health == TableHealth.Health.YELLOW) {
+                            finalDescription = "One or more tables have underreplicated shards";
```

Comment:
Do you think it would make sense to include the table names in the description, or would that get too long?

Given that the info can be retrieved via `sys.health` in a structured way it is probably not too important, but could be convenient.

Reply:
I wonder how much this helps, as if this occurs, one might need to look into `sys.health` to read the missing/underreplicated shards per each table anyway. But if you think it's helpful, I'm fine to add this (with truncation).

Reply:
We can also leave it as is for now, and extend it later if we notice that it would be useful to have the info.